### PR TITLE
Initial Docker page

### PR DESCRIPTION
**Pull request by donatieng**
Add docker user guide.

### CODE REVIEW

**Comment by donatieng** [comment]
Thanks Saheer, I cannot approve as I opened the original PR but it looks good :D

**Review by saheerb — APPROVED** [review]

File: docs/tools/docker/docker.md [hunk 1]
```diff
@@ -34,12 +34,14 @@ Run interactively in current directory and compile the current application:
...
 git clone https://github.com/ARMmbed/mbed-os-example-blinky.git && cd mbed-os-example-blinky
 
-docker run -i -t --mount=type=bind,source="$(pwd)",destination=/var/mbed,consistency=delegated -w /var/mbed ghcr.io/armmbed/mbed-os-env
+docker run -i -t --mount=type=bind,source="$(pwd)",destination=/var/mbed -w /var/mbed ghcr.io/armmbed/mbed-os-env
 
 mbed-tools deploy
 mbed-tools compile -t GCC_ARM -m DISCO_L475VG_IOT01A
 ```
 
+💡 When building the Mbed OS project inside Docker container with shared workspace from Docker host, there could be  performance issues as filesystem needs to be synced between Docker host and container. For better performance make sure, `gRPC FUSE for file sharing` [is enabled in Docker settings](https://www.docker.com/blog/deep-dive-into-new-docker-desktop-filesharing-implementation/).
```
Comment (ChrisGooch-Arm) [inline docs/tools/docker/docker.md:43]: is this meant to be a note?

Comment (ChrisGooch-Arm) [reply docs/tools/docker/docker.md:43]: should be "as the filesystem" (I can fix this in dev after merging)

Comment (saheerb) [reply docs/tools/docker/docker.md:43]: > is this meant to be a note?

This is meant to be a "tip".

Comment (saheerb) [reply docs/tools/docker/docker.md:43]: > should be "as the filesystem" (I can fix this in dev after merging)

Thanks, please fix as you mentioned.

**Pull request merged**